main/src/cgeo/geocaching/Image.java

```diff
@@ -15,27 +17,92 @@
 
 import java.io.File;
 
+/**
+ * Represent an Image along with Title and Description.
+ * Mostly used for representing image in/from logs.
+ */
 public class Image implements Parcelable {
-    private final String url;
-    private final String title;
-    private final String description;
+    @NonNull private Uri uri;
```

Comment:
Note that I don't insist on having final variables be `private`with a getter: a `public final` field is as safe as a `private final` one with a getter, since it can't be modified.